### PR TITLE
[SPARK-14119][SPARK-14120][SPARK-14122][SQL] Throw exception on unsupported DDL commands

### PR DESCRIPTION
**Pull request by andrewor14**
## What changes were proposed in this pull request?

Before: We just pass all role commands to Hive even though it doesn't work.
After: We throw an `AnalysisException` that looks like this:

```
scala> sql("CREATE ROLE x")
org.apache.spark.sql.AnalysisException: Unsupported Hive operation: CREATE ROLE;
  at org.apache.spark.sql.hive.HiveQl$$anonfun$parsePlan$1.apply(HiveQl.scala:213)
  at org.apache.spark.sql.hive.HiveQl$$anonfun$parsePlan$1.apply(HiveQl.scala:208)
  at org.apache.spark.sql.catalyst.parser.CatalystQl.safeParse(CatalystQl.scala:49)
  at org.apache.spark.sql.hive.HiveQl.parsePlan(HiveQl.scala:208)
  at org.apache.spark.sql.SQLContext.parseSql(SQLContext.scala:198)
```
## How was this patch tested?

`HiveQuerySuite`

### CODE REVIEW

**Comment by andrewor14** [comment]
@yhuai this should be straightforward

**Comment by rxin** [comment]
who's actually throwing the unsupported exception?

**Comment by rxin** [comment]
never mind it is catalystql. can you put the actual error message thrown into the pr description?

**Comment by andrewor14** [comment]
updated

**Comment by rxin** [comment]
Sorry for not being clear. I meant the entire exception. I'm wondering if there are any funny error showing up in the that exception.

**Comment by andrewor14** [comment]
updated

**Comment by yhuai** [comment]
Thanks for the PR! It will be great if we can make it clear what command is not supported. From the current error message, we only know there are features that we do not support. But, the error message does not say what is the unsupported command. Also, the current message is pretty verbose.

**Comment by andrewor14** [comment]
@yhuai if we want any special message we'll need to parse the tokens ourselves. We could do that but it requires more code. Any concrete suggestions in mind for the message?

**Comment by hvanhovell** [comment]
@andrewor14 I thought we passed these on to Hive and would execute these for us. Is this not working at?

As for better messaging: You could do some lightweight parsing on the AST nodes since the `TOK_*` names used in the toplevel node are usually quite informative (only a mapping would be required).

**Comment by yhuai** [comment]
@hvanhovell The thought is that it is better to all commands by Spark because asking hive to execute some commands may introduce inconsistent behaviors or failures with bad error messages. Also, we can reduce the dependence on hive at the execution side. For commands shown in this PR, since we do not support them right now, we just throw exceptions.

**Comment by andrewor14** [comment]
OK, @yhuai PTAL

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala [hunk 1]
```diff
@@ -209,6 +208,9 @@ private[hive] class HiveQl(conf: ParserConf) extends SparkQl(conf) with Logging
     safeParse(sql, ParseDriver.parsePlan(sql, conf)) { ast =>
       if (nativeCommands.contains(ast.text)) {
         HiveNativeCommand(sql)
+      } else if (hiveUnsupportedCommands.contains(ast.text)) {
+        val humanReadableText = hiveUnsupportedCommands(ast.text)
+        throw new AnalysisException("Unsupported Hive operation: " + humanReadableText)
```
Comment (rxin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala]: remove "Hive"; just "Unsupported operation"

**Comment by rxin** [comment]
LGTM

**Comment by andrewor14** [comment]
Thanks merging into master.

**Pull request closed without merging**